cluster/ci/origin/logging-infra/finished.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"time"
+
+	"os"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+)
+
+type Finished struct {
+	// Timestamp is the time at which we finish
+	// execution, in seconds past the UNIX epoch
+	Timestamp int64 `json:"timestamp"`
+
+	// Passed is the result of the build
+	Passed bool `json:"passed"`
+}
+
+func UploadFinishedData(processLog, metadataFile, artifactDir string, passed bool, gcsBucket *storage.BucketHandle) error {
+	metadata, err := loadMetadata(metadataFile)
+	if err != nil {
+		return err
+	}
+
+	finishedFile, err := writeFinishedMetadata(passed)
+	if err != nil {
+		return err
+	}
+
+	gcsPath := metadata.GcsPath()
+	uploadTargets := map[string]string{
+		finishedFile: gcsPath + "/finished.json",
+		// TODO(skuznets): we want to stream this log during the run
```

Comment:
Why? We want one place that knows how to upload from the test env to GCS or whatever intermediate storage is used to interface with other tools. Not sure centralizing that information to prow is correct